nilearn/glm/_base.py

```diff
@@ -546,6 +536,16 @@ def _generate_statistical_maps(
         # Override automatic detection with explicit type if provided
         stat_type = contrast_types.get(contrast_name, stat_type)
 
+        fields = {
+            "prefix": prefix,
+            "suffix": "statmap",
+            "extension": "nii.gz",
+            "entities": deepcopy(entities),
```

Comment:
Do we really need deep copy of entities? As far as I see, `entities` is just  used to extract filename. May be I'm missing something.

Reply:
this is mostly to avoid side effects as entities needs to be modified in the loop in this function but we don't want it to be modified out of it: without the copy, entities would also the be changed in other places in the caller function

once again maybe I'm just to dumb to understand classes, OOP and references but this solved my issue

Reply:
Not really related to OOP, but yes, I missed the fact that entities change, sorry.

Reply:
well in python everything is an object, so everything is OOP... 😜